monai/apps/auto3dseg/bundle_gen.py

```diff
@@ -175,46 +176,71 @@ def _create_cmd(self, train_params=None):
-
-        if "CUDA_VISIBLE_DEVICES" in params:
-            devices = params.pop("CUDA_VISIBLE_DEVICES")
-            n_devices, devices_info = len(devices), ",".join([str(x) for x in devices])
-        else:
-            n_devices, devices_info = torch.cuda.device_count(), ""
-        if n_devices > 1:
-            cmd = f"torchrun --nnodes={1:d} --nproc_per_node={n_devices:d} "
+        cmd = self.device_setting['CMD_PREFIX']
+        # make sure cmd end with a space
+        if cmd is not None and not cmd.endswith(' '):
+            cmd += ' '
+        if self.device_setting['NUM_NODES'] > 1:
+            # if using multinode
+            if self.device_setting['MN_START_METHOD'] == 'bcprun':
+                cmd = 'python ' if cmd is None else cmd
+            else:
+                raise NotImplementedError(f"{self.device_setting['MN_START_METHOD']} is not supported yet. \
+                                             Try modify BundleAlgo._create_cmd for your cluster.")
         else:
-            cmd = "python "  # TODO: which system python?
+            # if using single node
+            if self.device_setting['n_devices'] > 1:
+                cmd = f"torchrun --nnodes={1:d} --nproc_per_node={self.device_setting['n_devices']:d} " if cmd is None else cmd
+            else:
+                cmd = 'python ' if cmd is None else cmd
         cmd += base_cmd
         if params and isinstance(params, Mapping):
             for k, v in params.items():
                 cmd += f" --{k}={v}"
-        return cmd, devices_info
+        return cmd
 
-    def _run_cmd(self, cmd: str, devices_info: str) -> subprocess.CompletedProcess:
+    def _run_cmd(self, cmd: str) -> subprocess.CompletedProcess:
         """
         Execute the training command with target devices information.
 
         """
 
         logger.info(f"Launching: {cmd}")
         ps_environ = os.environ.copy()
-        if devices_info:
-            ps_environ["CUDA_VISIBLE_DEVICES"] = devices_info
-        normal_out = subprocess.run(cmd.split(), env=ps_environ, check=True)
-
+        ps_environ["CUDA_VISIBLE_DEVICES"] = self.device_setting['CUDA_VISIBLE_DEVICES']
+        if self.device_setting['NUM_NODES'] > 1:
+            if self.device_setting['MN_START_METHOD'] == 'bcprun':
+                normal_out = subprocess.run(['bcprun','-n',str(self.device_setting['NUM_NODES']),'-p',
+                                             str(self.device_setting['n_devices']),'-c', cmd], env=ps_environ, check=True)
+            else:
+                raise NotImplementedError(f"{self.device_setting['MN_START_METHOD']} is not supported yet.\
+                                             Try modify BundleAlgo._run_cmd for your cluster.")
+        else:
+            normal_out = subprocess.run(cmd.split(), env=ps_environ, check=True)
         return normal_out
 
-    def train(self, train_params=None):
+    def train(self, train_params: None | dict = None, device_setting: None | dict = None):
```

Comment:
The train function no longer takes "CUDA_VISIBLE_DEVICES" as an effective param in the "train_params", which seems another breaking change.

Reply:
I would suggest to raise deprecation warnings if user puts "CUDA_VISIBLE_DEVICES" in train_params and let them know it can be written in `device_setting`.

Reply:
After some thoughts, I think it is reasonable to drop the support `CUDA_VISIBLE_DEVICES` in `train_params`, because we don't want two different ways to change the GPU device settings. But it would be good to raise warning here to let user know it is no longer supported.

Reply:
I added this to set_train_params:
```
if 'CUDA_VISIBLE_DEVICES' is self.train_params.keys():
        warnings.warn("CUDA_VISIBLE_DEVICES definition is deprecated from `set_training_params`. Use `set_device_info` intead.", warnings.DeprecationWarning)
```